analytic_engine/src/memtable/columnar/mod.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic,
+        atomic::{AtomicU64, AtomicUsize, Ordering},
+        Arc, RwLock,
+    },
+};
+
+use arena::MonoIncArena;
+use bytes::Bytes;
+use common_types::{
+    column::Column, datum::Datum, row::RowGroupSplitter, schema::Schema, SequenceNumber,
+};
+use common_util::error::BoxError;
+use log::debug;
+use skiplist::Skiplist;
+use snafu::{ensure, OptionExt, ResultExt};
+
+use crate::memtable::{
+    columnar::iter::ColumnarIterImpl,
+    factory::Options,
+    iter::ReversedColumnarIterator,
+    key::{BytewiseComparator, KeySequence},
+    ColumnarIterPtr, Internal, InternalNoCause, InvalidPutSequence, MemTable, PutContext, Result,
+    ScanContext, ScanRequest,
+};
+
+pub mod factory;
+pub mod iter;
+
+pub struct ColumnarMemTable {
+    /// Schema of this memtable, is immutable.
+    schema: Schema,
+    memtable: Arc<RwLock<HashMap<String, Column>>>,
+    /// The last sequence of the rows in this memtable. Update to this field
+    /// require external synchronization.
+    last_sequence: AtomicU64,
+    row_num: AtomicUsize,
+    opts: Options,
+    memtable_size: AtomicUsize,
+}
+
+impl ColumnarMemTable {
+    fn memtable_size(&self) -> usize {
+        self.memtable
+            .read()
+            .unwrap()
+            .iter()
+            .map(|(_, column)| column.size())
+            .sum()
+    }
```

Comment:
This operation is still very time-consuming, and modifying it to compute in rows avoids the time-consuming nature of this function.

Reply:
Add `TODO`.